src/pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -173,15 +173,17 @@ def __check_training_step_requires_dataloader_iter(model: "pl.LightningModule")
     if is_param_in_hook_signature(training_step_fx, "dataloader_iter", explicit=True):
```

Comment:
I think it would be better to not call these 2 hooks under this condition. Then it is up to the user to manually call them if it's desired.

Reply:
I am also fine with this idea.

Reply:
I looked further into this. This would be possible but we still need to call it automatically for the strategies and callbacks. Most importantly, some of our builtin callbacks rely on this so it would be weird to not do the same for the LightningModule. Resolving